### PR TITLE
Added order functionality

### PR DESCRIPTION
**Pull request by nshtykh**

### CODE REVIEW

**Review by Alexander1iubchenko — APPROVED** [review]
Looks good to me

File: src/main/java/com/example/onlinebookstore/dto/order/PostOrderRequestDto.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.example.onlinebookstore.dto.order;
+
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
+import lombok.Data;
+
+@Data
+public class PostOrderRequestDto {
+    @NotNull
```
Comment (vovasalyha) [inline src/main/java/com/example/onlinebookstore/dto/order/PostOrderRequestDto.java]:
```suggestion
```

File: src/main/java/com/example/onlinebookstore/model/Order.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+@Table(name = "orders")
+@SQLDelete(sql = "UPDATE orders SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @ManyToOne
```
Comment (vovasalyha) [inline src/main/java/com/example/onlinebookstore/model/Order.java]: Do you really need it to be EAGER?

File: src/main/java/com/example/onlinebookstore/model/OrderItem.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+@Table(name = "order_items")
+@SQLDelete(sql = "UPDATE order_items SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
+public class OrderItem {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @ManyToOne
```
Comment (vovasalyha) [inline src/main/java/com/example/onlinebookstore/model/OrderItem.java]: Do you really need it to be EAGER?

File: src/main/java/com/example/onlinebookstore/model/OrderItem.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    @JoinColumn(name = "order_id", nullable = false)
+    private Order order;
+    @ToString.Exclude
+    @EqualsAndHashCode.Exclude
+    @ManyToOne
```
Comment (vovasalyha) [inline src/main/java/com/example/onlinebookstore/model/OrderItem.java]: Do you really need it to be EAGER?

File: src/main/java/com/example/onlinebookstore/service/order/OrderServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+                        "Can't find order with id " + id));
+        order.setStatus(Order.Status.valueOf(requestDto.getStatus()));
+        orderRepository.save(order);
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    @Transactional
+    public OrderResponseDto placeOrder(Long userId, PostOrderRequestDto requestDto) {
+        User user = userRepository.findById(userId)
```
Comment (vovasalyha) [inline src/main/java/com/example/onlinebookstore/service/order/OrderServiceImpl.java]: I'm not sure that you need to fetch a user from DB as you are fetching a shopping cart by userId anyway.

File: src/main/java/com/example/onlinebookstore/service/order/OrderServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    private Set<OrderItem> createOrderItems(ShoppingCart shoppingCart, Order order) {
+        Set<OrderItem> orderItems = new HashSet<>();
+
+        for (CartItem cartItem : shoppingCart.getCartItems()) {
+            Book book = cartItem.getBook();
+
+            OrderItem orderItem = new OrderItem();
+            orderItem.setOrder(order);
+            orderItem.setBook(book);
+            orderItem.setQuantity(cartItem.getQuantity());
```
Comment (vovasalyha) [inline src/main/java/com/example/onlinebookstore/service/order/OrderServiceImpl.java:114]: Please verify that if you add order Items with the same quantities they are both added to HashSet. Use case: add 2 items with the same quantity and see that there are 2 items in the set.

**Review by vovasalyha — CHANGES_REQUESTED** [review]
Looks good, but see a couple of comments.

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**